torchrl/modules/models/models.py

```diff
@@ -1034,3 +1035,27 @@ def forward(
 
         input = self.mlp(input)
         return self._lstm(input, hidden0_in, hidden1_in)
+
+
+class GRUNet(nn.Module):
+    """
+    MLP -> GRU -> MLP
+
+    """
+
+    def __init__(
+        self, mlp_input_kwargs: Dict, gru_kwargs: Dict, mlp_output_kwargs: Dict
+    ) -> None:
+        super().__init__()
+        assert mlp_input_kwargs["out_features"] == gru_kwargs["input_size"]
+        assert mlp_output_kwargs["in_features"] == gru_kwargs["hidden_size"]
+
+        self.mlp_in = MLP(**mlp_input_kwargs)
+        self.gru = nn.GRUCell(**gru_kwargs)
+        self.mlp_out = MLP(**mlp_output_kwargs)
+
+    def forward(self, inputs, hidden_state):
+        mlp_in = self.mlp_in(inputs)
+        hidden_state = self.gru(mlp_in, hidden_state)
```

Comment:
I expect that if you give a sequence as input this module will not behave as expected right?
Should we enforce that the input shapes match some requirement? Like no 3d input?
Problem with that 3d inputs are only an indirect indicator that someone is passing tensors that are time-stamped, it might as well be a 2d batch size. Plus, a 1d batch size might be time-stamped too :p 

Perhaps a good docstring would go a long way. Can we put in the docstring of this class what kind of input we expect?

Reply:
I never worked with GRU that takes full sequences... It appears that I missed a nice feature.
Indeed, the current implementation will not work well when taking a sequence as input.

I'll rework this as it is done in the GRU, and using GRU instead of GRUCell then (enforcing batch first)

> **input**: tensor of shape :math:`(L, H_{in})` for unbatched input,
          :math:`(N, L, H_{in})` when ``batch_first=True`` containing the features of
          the input sequence.

This should lead to something closer to the LSTMNet already implemented.